File: Sliding_Window/1542-consecutive-characters/consecutive-characters.java
class Solution {
    public int maxPower(String s) {
        int maxLen = 0;
        int n = s.length();

        for (int i = 0; i < n;) {
            char cur = s.charAt(i); // Current character
            int j = i, len = 0;

            // Count consecutive identical characters
            while (j < n && s.charAt(j) == cur) {
                len++;
                j++;
            }

            i = j; // Move 'i' to the next non-matching character
            maxLen = Math.max(maxLen, len);
        }

        return maxLen;
    }
}

File: Sliding_Window/1542-consecutive-characters/consecutive-characters.py
class Solution:
    def maxPower(self, s: str) -> int:
        maxLen = 0
        n = len(s)

        i = 0
        while i < n:
            cur = s[i]  # Current character
            j = i
            length = 0

            # Count consecutive identical characters
            while j < n and s[j] == cur:
                length += 1
                j += 1

            i = j  # Move 'i' to the next non-matching character
            maxLen = max(maxLen, length)

        return maxLen
